proposed/container-configuration-meta.md

```diff
@@ -29,10 +29,70 @@ The way a framework can automatically **discover the container definitions** of
 
 ## 4. Approaches
 
-### 4.1. The format
+### 4.1. Features
 
-TODO
+First of all, a word of caution. Container configuration is shared between the application developer and the packages bringing some configuration. Ultimately, container configuration is the responsibility of the application developer. Packages should however be able to bring a "default" configuration to help the developer getting started with sensible defaults (container configuration can be quite complex sometimes). This configuration should be both optional (possibility to bypass it completely), and overwritable (possibility to changes bits of the configuration).
+
+In this section, we will describe the list of features that we can consider to fulfill the goal of this PSR: packages registering container entries. Let's list all possible features and later, we can vote on the features we want to keep or not.
+
+1. Ability to create a container entry using the `new` keyword.
+
+1. Ability to call methods (setters or otherwise) on a container entry.
+
+1. Ability to set public properties of a container entry.
+
+1. Ability to set private and protected properties of a container entry.
+
+1. Ability to create a container entry using a static factory method.
+
+1. Ability to create a container entry using a factory method from a container service.
+
+1. Ability to create a container entry using a closure.
+
+1. Ability to compile all container entries into a single container for maximum performance.
+
+1. Ability to alias a container entry to another.
+
+1. Ability to modify an entry defined outside of the "module" before it is returned by the container. For instance: ability for a package providing a `Twig_Extension` to modify the `Twig_Environment` by calling `$environment->addExtension($extension)`, even if the `Twig_Environment` entry was not declared by the package itself.
+
+1. Ability to create container entries for scalar values.
+
+1. Ability to create container entries for constants (from the `define` keyword or the `const` keyword).
+
+1. Ability to create container entries that are numerically indexed arrays. Values of the array can be any valid container entry (i.e. objects, scalars, another array...)
 
-### 4.2. Features
+1. Ability to create container entries that are associative arrays (maps). Values of the array can be any valid container entry (i.e. objects, scalars, another array...)
```

Comment:
> You mean I should split "array of services" and "array of scalar values" in 2 different features?

Definitely IMO. Many containers don't support the array of services. "Array of scalar values" could even go in the "Ability to define scalar values" (because if it's just a serializable value then we don't care if it's a int, string, array of int, etc.).